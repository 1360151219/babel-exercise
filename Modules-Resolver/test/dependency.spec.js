import { expect, test } from 'vitest'
import path from 'path'
import getDependencyGraph from '../index.js'
function add(a, b) {
  return a + b
}

test('single module graph', () => {
  const graph = JSON.stringify(getDependencyGraph(path.resolve(__dirname, './module1.js')), null, 4)
  console.log(graph);
  expect(graph).toMatchSnapshot('dependencyGraph')
})

