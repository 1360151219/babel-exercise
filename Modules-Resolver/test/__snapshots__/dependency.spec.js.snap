// Vitest Snapshot v1

exports[`single module graph > dependencyGraph 1`] = `
"{
    \\"root\\": {
        \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module1.js\\",
        \\"imports\\": {
            \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module2.js\\": [
                {
                    \\"local\\": \\"a\\",
                    \\"imported\\": \\"a\\",
                    \\"type\\": \\"deconstruct\\"
                },
                {
                    \\"local\\": \\"bModule2\\",
                    \\"imported\\": \\"b\\",
                    \\"type\\": \\"deconstruct\\"
                }
            ]
        },
        \\"exports\\": [],
        \\"subModules\\": {
            \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module2.js\\": {
                \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module2.js\\",
                \\"imports\\": {
                    \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\": [
                        {
                            \\"local\\": \\"c\\",
                            \\"type\\": \\"default\\"
                        }
                    ]
                },
                \\"exports\\": [
                    [
                        {
                            \\"local\\": \\"a\\",
                            \\"exported\\": \\"a\\",
                            \\"type\\": \\"named\\"
                        },
                        {
                            \\"local\\": \\"b\\",
                            \\"exported\\": \\"b\\",
                            \\"type\\": \\"named\\"
                        }
                    ]
                ],
                \\"subModules\\": {
                    \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\": {
                        \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\",
                        \\"imports\\": {
                            \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": [
                                {
                                    \\"local\\": \\"d\\",
                                    \\"type\\": \\"namespace\\"
                                }
                            ]
                        },
                        \\"exports\\": [
                            [
                                {
                                    \\"local\\": \\"c\\",
                                    \\"exported\\": \\"ccc\\",
                                    \\"type\\": \\"named\\"
                                }
                            ]
                        ],
                        \\"subModules\\": {
                            \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": {
                                \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\",
                                \\"imports\\": {},
                                \\"exports\\": [
                                    {
                                        \\"type\\": \\"default\\",
                                        \\"exported\\": \\"1\\"
                                    }
                                ],
                                \\"subModules\\": {}
                            }
                        }
                    }
                }
            }
        }
    },
    \\"allModules\\": {
        \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module1.js\\": {
            \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module1.js\\",
            \\"imports\\": {
                \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module2.js\\": [
                    {
                        \\"local\\": \\"a\\",
                        \\"imported\\": \\"a\\",
                        \\"type\\": \\"deconstruct\\"
                    },
                    {
                        \\"local\\": \\"bModule2\\",
                        \\"imported\\": \\"b\\",
                        \\"type\\": \\"deconstruct\\"
                    }
                ]
            },
            \\"exports\\": [],
            \\"subModules\\": {
                \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module2.js\\": {
                    \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module2.js\\",
                    \\"imports\\": {
                        \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\": [
                            {
                                \\"local\\": \\"c\\",
                                \\"type\\": \\"default\\"
                            }
                        ]
                    },
                    \\"exports\\": [
                        [
                            {
                                \\"local\\": \\"a\\",
                                \\"exported\\": \\"a\\",
                                \\"type\\": \\"named\\"
                            },
                            {
                                \\"local\\": \\"b\\",
                                \\"exported\\": \\"b\\",
                                \\"type\\": \\"named\\"
                            }
                        ]
                    ],
                    \\"subModules\\": {
                        \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\": {
                            \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\",
                            \\"imports\\": {
                                \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": [
                                    {
                                        \\"local\\": \\"d\\",
                                        \\"type\\": \\"namespace\\"
                                    }
                                ]
                            },
                            \\"exports\\": [
                                [
                                    {
                                        \\"local\\": \\"c\\",
                                        \\"exported\\": \\"ccc\\",
                                        \\"type\\": \\"named\\"
                                    }
                                ]
                            ],
                            \\"subModules\\": {
                                \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": {
                                    \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\",
                                    \\"imports\\": {},
                                    \\"exports\\": [
                                        {
                                            \\"type\\": \\"default\\",
                                            \\"exported\\": \\"1\\"
                                        }
                                    ],
                                    \\"subModules\\": {}
                                }
                            }
                        }
                    }
                }
            }
        },
        \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module2.js\\": {
            \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module2.js\\",
            \\"imports\\": {
                \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\": [
                    {
                        \\"local\\": \\"c\\",
                        \\"type\\": \\"default\\"
                    }
                ]
            },
            \\"exports\\": [
                [
                    {
                        \\"local\\": \\"a\\",
                        \\"exported\\": \\"a\\",
                        \\"type\\": \\"named\\"
                    },
                    {
                        \\"local\\": \\"b\\",
                        \\"exported\\": \\"b\\",
                        \\"type\\": \\"named\\"
                    }
                ]
            ],
            \\"subModules\\": {
                \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\": {
                    \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\",
                    \\"imports\\": {
                        \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": [
                            {
                                \\"local\\": \\"d\\",
                                \\"type\\": \\"namespace\\"
                            }
                        ]
                    },
                    \\"exports\\": [
                        [
                            {
                                \\"local\\": \\"c\\",
                                \\"exported\\": \\"ccc\\",
                                \\"type\\": \\"named\\"
                            }
                        ]
                    ],
                    \\"subModules\\": {
                        \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": {
                            \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\",
                            \\"imports\\": {},
                            \\"exports\\": [
                                {
                                    \\"type\\": \\"default\\",
                                    \\"exported\\": \\"1\\"
                                }
                            ],
                            \\"subModules\\": {}
                        }
                    }
                }
            }
        },
        \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\": {
            \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module3.js\\",
            \\"imports\\": {
                \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": [
                    {
                        \\"local\\": \\"d\\",
                        \\"type\\": \\"namespace\\"
                    }
                ]
            },
            \\"exports\\": [
                [
                    {
                        \\"local\\": \\"c\\",
                        \\"exported\\": \\"ccc\\",
                        \\"type\\": \\"named\\"
                    }
                ]
            ],
            \\"subModules\\": {
                \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": {
                    \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\",
                    \\"imports\\": {},
                    \\"exports\\": [
                        {
                            \\"type\\": \\"default\\",
                            \\"exported\\": \\"1\\"
                        }
                    ],
                    \\"subModules\\": {}
                }
            }
        },
        \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\": {
            \\"path\\": \\"/Applications/workplace/babel-exercise/Modules-Resolver/test/module4.js\\",
            \\"imports\\": {},
            \\"exports\\": [
                {
                    \\"type\\": \\"default\\",
                    \\"exported\\": \\"1\\"
                }
            ],
            \\"subModules\\": {}
        }
    }
}"
`;
